library/ratelimiting/shouldRateLimitRequest.ts

```diff
@@ -21,6 +21,15 @@ export function shouldRateLimitRequest(
   context: Readonly<Context>,
   agent: Agent
 ): Result {
+  // Do not consume rate limit for the same request a second time
+  // (Might happen if the user adds the middleware multiple times)
+  if (context.consumedRateLimit) {
```

Comment:
should we trigger a single time warning? like we did for setUser call?

Reply:
Added